cps2pc

```diff
@@ -0,0 +1,226 @@
+            prefix = _compilers_definition_prefixes[compiler][ma.groups()[0]]
+            definitions.append(prefix + ma.groups()[1])
+        except KeyError:
+            # This should never happen as the list of supported compilers
+            # is hard-coded and the definitions either start with `!` or don't.
+            # If we are here, there is a bug.
+            raise Exception('Problem reading definition argument `%s`. '
+                            'Most likely a bug in this script.' % definition)
+    return definitions
+
+#------------------------------------------------------------------------------
+def get_platform(package):
+    try:
+        return package.platform.kernel
+    except AttributeError:
+        try:
+            output = subprocess.check_output(['uname', '-s'])
+            return output.strip().lower()
+        except OSError:
+            raise Exception('Unable to detect current platform')
+
+#------------------------------------------------------------------------------
+def get_kind_pc_dictionary(kind, location):
+    kind_dict = {}
+    if kind == 'exe':
+        raise Exception('executable')
+    elif kind in ['dylib', 'static'] and location:
+        # Full path is specified directly without any flag.
+        kind_dict['Libs'] = [' %s' % location]
+    elif kind == 'jar':
+        kind_dict['classpath'] = location
+    elif kind == 'interface':
+        pass
+    else:
+        raise Exception(kind)
+    return kind_dict
+
+#------------------------------------------------------------------------------
+def write_pc(pc_dict, filename, prefix):
+    def list_to_str(l):
+        return ' '.join(l)
+    convert = {'unicode': unicode, 'str': str, 'list': list_to_str}
+    with open(filename, 'w') as f:
+        for k, v in pc_dict.iteritems():
+            f.write('%s: %s\n' % (k, convert[type(v).__name__](v).replace(
+                '@prefix@', prefix)))
+
+#------------------------------------------------------------------------------
+def remove_empty_keys(pc_dict):
+    for k, v in pc_dict.iteritems():
+        if not v:
+            pc_dict.pop(k)
+
+#------------------------------------------------------------------------------
+def optional(extensions, attr):
+    try:
+        return extensions.data[attr]
+    except KeyError:
+        return []
+
+#------------------------------------------------------------------------------
+def write_targets_config(package, output_dir, compiler, prefix):
+    platform = get_platform(package)
+    for component_name, component in package.components.iteritems():
+        # Ignored CPS tags (not supported by pkgconfig):
+        # * Link-Languages
+        # * Configurations
+        pc_dict = OrderedDict()
+        pc_dict['Name'] = component_name
+        pc_dict['Version'] = package.version
+
+        # Handle supplemental schema
+        for element in ['Description', 'Website']:
+            pc_dict[element] = optional(package, element)
```

Comment:
The RHS can be replaced by `package.data.get(element, [])`? (And drop the unnecessary `optional` function?) I'm a little confused though why `[]` is used as a fall-back; if the attribute is present, won't it be a string? Wouldn't `None` make more sense then as the fall-back? And in that case, do we even need to use `get`, or should this just be:
```python
        for element in ['Description', 'Website']:
            if element in package.data:
                pc_dict[element] = package.data[element]
```
...? (But if you prefer to rely on `remove_empty_keys` to nuke empty values later, that's okay too.)